benchmarks/sparse/dlmc/spmm.py

```diff
@@ -0,0 +1,150 @@
+# and with other frameworks such as scipy. 
+
+import sys
+from pathlib import Path
+import pandas as pd
+import argparse
+import torch
+import torch.utils.benchmark as benchmark_utils
+from .utils import load_spmm_dataset, to_coo_scipy, scipy_coo_matmul
+import pickle
+
+def torch_backward(a_dense, b_dense):
+    a_dense.requires_grad = True
+    b_dense.requires_grad = True
+    r1 = a_dense.matmul(b_dense)
+    f1 = torch.sum(r1)
+    f1.backward()
+
+
+def sparse_torch_backward(a, b):
+    a.requires_grad = True
+    b.requires_grad = True
+
+    r2 = torch.sparse.mm(a, b)
+    if r2.is_sparse:
+        f2 = torch.sparse.sum(r2)
+    else:
+        f2 = torch.sum(r2)
+    f2.backward()
+
+
+if __name__ == '__main__':
+
+    path = Path()
+    parser = argparse.ArgumentParser(description='spmm Bench')
+
+    parser.add_argument('--path', type=str, help='dataset path')
+    parser.add_argument('--dataset',
+                        type=str,
+                        help='dataset name (random_pruning, variational_dropout, magnitude_pruning, \
+                            extended_magnitude_pruning)',
+                        default='random_pruning')
+    parser.add_argument('--operation',
+                        type=str,
+                        help='matmul or backward',
+                        default='matmul')
+    parser.add_argument('--spmm_type',
+                        type=str,
+                        help='sparse-sparse or sparse-dense',
+                        default='sparse-dense')
+    parser.add_argument('--output',
+                        type=str,
+                        help='dataframe output path',
+                        default='/tmp/matmul_bench.pkl')
+    parser.add_argument('--device',
+                        type=str,
+                        help='device (cpu or cuda)',
+                        default='cpu')
+
+    args = parser.parse_args()
+    print('path     =', args.path)
+    print('dataset       =', args.dataset)
+    print('spmm_type        =', args.spmm_type)
+    print('operation     =', args.operation)
+    print('output        =', args.output)
+    print('device        =', args.device)
+
+    dataset_path = args.path
+    dataset_name = args.dataset
+    dataset_path = f"{dataset_path}/{dataset_name}"
+    df_output_path = args.output
+    spmm_type = args.spmm_type
+    device = args.device
+
+    tasks = []
+    if args.operation == 'matmul':
+        if device == 'cpu':
+            tasks = [
+                ("matmul", device, "torch", "torch.mm(dx, dy)"),
+                ("matmul", device, "torch.sparse", "torch.sparse.mm(x, y)"),
+                ("matmul", device, "scipy", "scipy_coo_matmul(sx, sy)"),
+            ]
+        else:
+            tasks = [
+                ("matmul", device, "torch", "torch.mm(dx, dy)"),
+                ("matmul", device, "torch.sparse", "torch.sparse.mm(x, y)"),
+            ]
+    else:
```

Comment:
check that it's supported value, otherwise any unexpected `operation` value would be interpreted as `backward`